### PR TITLE
Feature/ele 582 section sector picker should use real data

### PR DESCRIPTION
**Pull request by dannelundqvist**
Replaced all uses of (hardcoded) section default values with cached data fetched from the index.

This meant we lost "color dots" for sections in tables. That data does not exist in the index. We should discuss that as a separate issue. (I left the defaults files as deprecated reference for this reason)

### CODE REVIEW

File: src/lib/index/authors.ts
Comment (gularsson) [inline src/lib/index/authors.ts:1]: `authors.ts`, `sections.ts` and `stories.ts` are close to identical. One function? Could maybe just use `get.ts`?

File: src/lib/index/stories.ts
Comment (gularsson) [inline src/lib/index/stories.ts:1]: See `authors.ts`

File: src/datastore/contexts/CoreSectionProvider.tsx
Comment (gularsson) [inline src/datastore/contexts/CoreSectionProvider.tsx:1]: Refactor? See PR comment at top

File: src/hooks/useSections.tsx
Comment (gularsson) [inline src/hooks/useSections.tsx:1]: Refactor? See PR comment above.

File: src/views/Planning/components/PlanSection.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+export const PlanSection = (): JSX.Element => {
+  const allStories = useSections().map((_) => {
+    return {
+      value: _.id,
+      label: _.title
+    }
+  })
+
+  const [, setStories] = useYValue<Y.Array<unknown>>('links.core/section', {
```
Comment (gularsson) [inline src/views/Planning/components/PlanSection.tsx:16]: `setStories`? Should be `setSections` right?

File: src/views/Planning/components/PlanSection.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
+import { Awareness } from '@/components'
+import { ComboBox } from '@/components/ui'
+import { useSections, useYValue } from '@/hooks'
+import { Block } from '@/protos/service'
+import { useRef } from 'react'
+import * as Y from 'yjs'
+
+export const PlanSection = (): JSX.Element => {
+  const allStories = useSections().map((_) => {
```
Comment (gularsson) [inline src/views/Planning/components/PlanSection.tsx:9]: `allSections`

**Review by gularsson — CHANGES_REQUESTED** [review]
Over all looks good, great to finally get proper data to work with and get rid of the mocked/static data

I cant help to wonder if we shouldn't start to refactor this immediately. It's easy for me to say coming from an outside perspective :slightly_smiling_face: 

`CoreSectionProvider`, `CoreAuthorProvider` and `CoreStoryProvider` are (close to?) identical.

`useAuthor`, `useSection` and `useStory` could be one hook.
```
const { authors, section, story } = useIndexedData()
```
I dont see any point in keeping them as separate contexts or hooks in terms of rerenders or codewise.

**Comment by Victor-Lindh** [comment]
Besides the points that @gularsson already brought up above, I have nothing further to add.
Looks great otherwise!

**Comment by dannelundqvist** [comment]
> Over all looks good, great to finally get proper data to work with and get rid of the mocked/static data
> 
> I cant help to wonder if we shouldn't start to refactor this immediately. It's easy for me to say coming from an outside perspective 🙂

Totally agree, over the course of doing refactoring the originally messy code clearer patterns have emerged. On it. It will saves us tons of time later on.

File: src/datastore/lib/fetchSections.ts
[the remark this answers is not in the document]
Comment (gularsson) [reply src/datastore/lib/fetchSections.ts:1]: I saw that we lost the section colors, a specific section "transformer" could augment the data received from the index.

File: src/datastore/contexts/CoreSectionProvider.tsx
Comment (dannelundqvist) [reply src/datastore/contexts/CoreSectionProvider.tsx:1]: I will leave this as is for now.

File: src/hooks/useSections.tsx
Comment (dannelundqvist) [reply src/hooks/useSections.tsx:1]: Will leave as useSections is not the same logic as useAuthors for example.

**Pull request merged**